### DISCUSSION TITLE
Small addition to allow instantiate to build cmdstanr-based packages without install as source

### DISCUSSION THREAD

**Discussion by stemangiola** [wlandau/instantiate#25] [Ideas]
Hello,

I have prepared a pull request that enables `instantiate` to build cmdstanr-based packages without requiring users to install the package from the source (```install.packages(..., type = "source")```). This enhancement allows packages like sccomp (linked below) to be installed and compiled just like any other R package, simplifying the installation process for end-users.

https://github.com/MangiolaLaboratory/sccomp/tree/cmdstanr

For example, in the sccomp package, I have implemented the following:

	•	The models are compiled once and cached when needed, avoiding the need to install from the source.
	•	The process leverages several utilities from instantiate to handle the compilation efficiently, including stan_package_model() and stan_package_compile().

The PR would be very light, from here

https://github.com/stemangiola/instantiate

This would be the only simple change

<img width="1245" alt="image" src="https://github.com/user-attachments/assets/c990cdf9-ced9-45ac-814f-6c03e9082fd5">

This is how I am using this edited function

```r
    mod = instantiate::stan_package_model(
      name = name, 
      package = "sccomp", 
      compile = TRUE,
      cpp_options = list(stan_threads = TRUE)
    )
 ```

Thanks for considering this PR

**Comment by wlandau** [comment]
This seems similar to the approach `torch` takes, where files are added to the installed package file system after installation. Sounds like a good optional feature. Happy to look at a PR. I think parts of the README and DESCRIPTION may need to be modified slightly.

**Reply by stemangiola** [reply]
Sure, feel free to let me know what parts could be tweaked further.
